src/main/java/io/nats/client/SubscribeOptions.java

```diff
@@ -80,7 +80,7 @@ protected SubscribeOptions(Builder builder, boolean isPull, boolean isOrdered, S
                     throw JsSoOrderedReplicasNotSuppliedOrOne.instance();
                 }
                 Duration ccHb = builder.cc.getIdleHeartbeat();
-                if (ccHb != null && ccHb.toMillis() > hb) {
+                if (ccHb != null) {
```

Comment:
Changed to accept whatever the user provides, as long as they provide something, in which case the default will be used. This matches the go client behavior.